snapcraft/internal/meta.py

```diff
@@ -122,10 +123,15 @@ def _setup_from_setup(self):
 
         gui_src = os.path.join(setup_dir, 'gui')
         gui_dst = os.path.join(self.meta_dir, 'gui')
-        if os.path.exists(gui_dst) and os.path.exists(gui_src):
-            shutil.rmtree(gui_dst)
         if os.path.exists(gui_src):
-            shutil.copytree(gui_src, gui_dst)
+            for f in os.listdir(gui_src):
+                if os.path.splitext(f)[1] == '.desktop':
+                    logger.warning(
+                        "DEPRECATED: use the 'desktop' key for apps instead "
```

Comment:
I wouldn't necessarily deprecate this, new projects might not care about existing conventions.